### PR TITLE
Improve ghmerge ref handling and documentation

### PR DESCRIPTION
**Pull request by DDvO**
* `ghmerge`: Make pulling the latest REMOTE/REF work als for non-default REF
    
    This avoids potential git errors such as:
    
    You asked to pull from the remote 'upstream', but did not specify
    a branch. Because this is not the default configured remote
    for your current branch, you must specify a branch on the command line.

* `ghmerge`: improve doc of `--remote` and `--ref` options

* `addrev`: re-enable use of singe-character review names

### CODE REVIEW

**Review by t8m — APPROVED** [review]
LGTM

**Comment by DDvO** [comment]
> LGTM

Thanks @t8m for having a look and approving.

I just found that there was also an issue with using `addrev` with giving PR numbers without the (optional) `--prnum=` prefix.
So I've added a commit fixing this:
* `addrev`: Fix regression on parsing bare prnum; correct help output

File: review-tools/addrev
Comment (t8m) [inline review-tools/addrev:19]: indent by 2 more spaces?

File: review-tools/addrev
Comment (t8m) [inline review-tools/addrev:17]: I understand that you want to support all numeric 7 character git commit id. Hopefully the PR numbers won't be in millions range too soon. :grin:

**Review by t8m — APPROVED** [review]

File: review-tools/addrev
Comment (DDvO) [reply review-tools/addrev]: This was due to a tab char. 
It turns out there were many of them in that file, which I've just eliminated.

File: review-tools/addrev
Comment (DDvO) [reply review-tools/addrev:17]: Yes. Recently the PR numbers became 5-digit, and it will certainly take a while until 6 digits are needed
By that time likely also more than 7 hex digits will be needed to identify commits,
so then our (grand)children can simultaneously increment both limits by one :sweat_smile:

**Review by t8m — APPROVED** [review]

**Comment by DDvO** [comment]
I think a 2nd approval is needed here, and I cannot set any labels to indicate that...

**Comment by t8m** [comment]
I don't know if it is needed. Do there have to be two OTC member approvals on tools? I did not think so.

**Review by paulidale — APPROVED** [review]

**Review by newblaque — APPROVED** [review]

**Comment by DDvO** [comment]
Can an OTC member please merge this?

BTW, would be nice if committers got privilege for that, too.

**Comment by t8m** [comment]
Merged now.

**Comment by t8m** [comment]
Aargh. I've used --tools with ghmerge, but apparently it did not pass the option to addrev so the references in the commits are wrong. Unfortunately it is already pushed.

**Comment by DDvO** [comment]
Good spotting.
I just found this was not the first and only glitch like that:
```
git log|grep github.com/openssl/openssl|wc
```
on the tools master gives 50 hits since 2017.

Yet another item to improve on `ghmerge` :)
I never came across this because i cannot merge to the tools repo.
Since I cannot really test it, I'm leaving this bug for someone else to fix.

**Comment by t8m** [comment]
#80

**Pull request closed without merging**
